kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/model/VirtualCluster.java

```diff
@@ -160,7 +160,7 @@ private Optional<SslContext> buildDownstreamSslContext() {
     private Optional<SslContext> buildUpstreamSslContext() {
         return targetCluster.tls().map(tls -> {
             try {
-                var sslContextBuilder = SslContextBuilder.forClient();
+                var sslContextBuilder = tls.key().forClient();
```

Comment:
I think this is wrong as it assumes client will always present a client certificate.

I think the formulation you want will look like this:

```suggestion
                var sslContextBuilder = Optional.ofNullable(tls.key()).map(KeyProvider::forClient).orElse(SslContextBuilder.forClient());
```